### ISSUE TITLE
[Question] All requests are GET, what about Post?

### ISSUE THREAD

**Issue by amrro** [googlesamples/android-architecture-components#102]
**TL;DR**: How to use `POST`instead of `GET`?

Thanks for your effort, really great learning experience browsing your code. But I have small question.

What I've noticed that all requests in [`GithubService`](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/api/GithubService.java) are `GET` type. By using the abstract class [`NetworkBoundResource`](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.java) to load each object by overriding `createCall()` in its Repository, like [`UserRepository#loadUser(String login)`](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/UserRepository.java#L49-L73). 

In `Post`, I update data remotely on server which I might have instance of it locally with Room, What/How to update Room depending on the API? Is the same class, [`NetworkBoundResource`](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.java), is applicable with `POST` requests? What is the best scenario in this case?

Thanks!

**Comment by SergeiVasilenko** [comment]
[NetworkBoundResources](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.java) is not applicable with `POST` request for me. I created similar class with follow difference:

```
@MainThread
ChangeAction(AppExecutors appExecutors) {
    mAppExecutors = appExecutors;
    makeChange();
}

private void makeChange() {
    LiveData<ApiResponse<RequestType>> apiResponse = createCall();
    apiResponse.observeForever(response -> {
        if (response.isSuccessful()) {
            mAppExecutors.diskIO().execute(() -> {
                saveCallResult(processResponse(response));
                mAppExecutors.mainThread().execute(() -> {
                    mResult.addSource(loadFromDb(), newData -> mResult.setValue(Resource.success(newData)));
                });
            });
        } else {
            onChangeFailed();
            mResult.addSource(loadFromDb(), newData -> mResult.setValue(Resource.error(response.error, newData)));
        }
    });
}
```

Be careful! I didn't test this code, and it may have bugs.

**Comment by amrro** [comment]
I suggest to edit [`saveCallResult(@NonNull RequestType item)`](https://github.com/googlesamples/android-architecture-components/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.java#L98) instead of adding more code. So, every Call in any repository handle its case.

**Comment by beylmk** [comment]
@amrro could you elaborate on your response or provide an example?

**Comment by plabon** [comment]
What is the solution to this post request problem? Is there any google recommendation?

**Comment by dlam** [comment]
@plabon Similar to GithubService linked earlier, OkHttp/Retrofit also provides an `@POST` annotation.

**Comment by plabon** [comment]
@dlam No i was asking about using [NetworkBoundResources](https://github.com/android/architecture-components-samples/blob/178fe541643adb122d2a8925cf61a21950a4611c/GithubBrowserSample/app/src/main/java/com/android/example/github/repository/NetworkBoundResource.java) How can I put data in local database and post the data to server?
